lib/eqaf.ml

```diff
@@ -111,3 +111,65 @@ let equal a b =
   if (al lxor ln) lor (bl lxor ln) <> 0
```

Comment:
Isn't this basically the same as this?
```ocaml
if al <> bl
 then false
 else equal ~ln a b
```

In that case I think we can just do this ?
```ocaml
let al = String.length a in
let bl = String.length b in
if al <> bl then false else equal ~ln:al a b
```
(Then we also don't have to bother with implementing branch-free `min` :D )